SensorsAnalyticsSDK/src/main/java/com/sensorsdata/analytics/javasdk/SensorsAnalytics.java

```diff
@@ -1065,18 +1081,16 @@ String getHttpContent() {
       final String httpContent;
     }
 
-    HttpConsumer(String serverUrl, Map<String, String> httpHeaders) {
+    HttpConsumer(String serverUrl, Map<String, String> httpHeaders, CloseableHttpClient httpClient) {
       this.serverUrl = serverUrl.trim();
       this.httpHeaders = httpHeaders;
       this.compressData = true;
+      this.httpClient = httpClient;
     }
 
     synchronized void consume(final String data) throws IOException, HttpConsumerException {
       HttpUriRequest request = getHttpRequest(data);
       CloseableHttpResponse response = null;
-      if (httpClient == null) {
```

Comment:
`HttpConsumer` is never null. It is a private constructor, and all usages provide values.